src/input_output.jl

```diff
@@ -224,11 +224,21 @@ function band_data_to_dict!(dict, band_data::NamedTuple; save_ψ=false, save_ρ=
     end
 
     function gather_and_store!(dict, key, basis, data)
+        # Gather from all k-points, even possibly duplicated ones
         gathered = gather_kpts_block(basis, data)
         if !isnothing(gathered)
-            n_kpoints = length(basis.kcoords_global)
+            n_kpoints = basis.n_irreducible_kpoints
             n_spin    = basis.model.n_spin_components
-            dict[key] = reshape(gathered, (size(data[1])..., n_kpoints, n_spin))
+            n_kpt_tot = length(basis.kcoords_global)
+
+            dict[key] = reshape(gathered, (size(data[1])..., n_kpt_tot, n_spin))
+
+            # Only store irreducible k-points (assumed to be first in an array)
+            if n_kpt_tot > n_kpoints
+                index = ntuple(_ -> Colon(), ndims(dict[key]))
+                index = Base.setindex(index, 1:n_kpoints, ndims(dict[key]) - 1)
+                dict[key] = dict[key][index...]
```

Comment:
I don't like that you change the thing in the dict after writing. For me once things are put into the dict they are "final". I'd suggest to introduce a temporary variable to hold the reshaped array and apply the additional masking afterwards.